openmc_dagmc_wrapper/tallies/mesh_tallies/mesh_tally_2D.py

```diff
@@ -0,0 +1,123 @@
+from typing import Iterable, List, Tuple, Union
+
+import openmc
+
+# TODO:
+# i don't understand why . here works but not openmc_dagmc_wrapper
+from . import MeshTally3D
+
+
+class MeshTally2D(MeshTally3D):
+    """[summary]
+
+    Args:
+        tally_type (str): [description]
+        plane (str): "xy", "xz", "yz"
+        mesh_resolution (list): [description]
+        bounding_box ([type], optional): either a .h5m filename or
+            [point1, point2]. Defaults to None.
+    """
+
+    def __init__(
+        self,
+        tally_type: str,
+        plane: str,
+        bounding_box: List[Tuple[float]],
+        plane_slice_location: Tuple[float, float] = (1, -1),
+        mesh_resolution: Tuple[float, float] = (400, 400),
+    ):
+        self.plane = plane
+        self.plane_slice_location = plane_slice_location
+
+        super().__init__(tally_type, bounding_box, mesh_resolution)
+        self.name = self.tally_type + "_on_2D_mesh_" + self.plane
+
+    def create_mesh(self, mesh_resolution, bounding_box):
```

Comment:
This is a real asset imo. But maybe rather than having a `MeshTally2D` class, we could have a `RegularMesh2D` class inheriting from `openmc.RegularMesh`.

Note: if we use `openmc.CylindricalMesh` to create this `RegularMesh2D`, it will be better for revolution solids (like tokamak).